core/include/detray/utils/tuple_helpers.hpp

```diff
@@ -65,11 +90,18 @@ template <int N, typename... value_types>
 struct tuple_element<N, thrust::tuple<value_types...>>
     : thrust::tuple_element<N, thrust::tuple<value_types...>> {};
 
+// detray::tuple
+template <int N, typename... value_types>
```

Comment:
I see that the other classes also use `int` for this type. But they all should really rather be using `std::size_t`...

Reply:
Fair, I just did this to keep it in line with the rest. :stuck_out_tongue: